src/generators/legacy-json/utils/buildSection.mjs

```diff
@@ -0,0 +1,314 @@
+        if (type) section.propertySigType = type;
+        Object.assign(section, rest);
+        section.textRaw = `\`${section.name}\` ${section.textRaw}`;
+      }
+      break;
+    case 'event':
+      section.params = values;
+      break;
+    default:
+      if (list) nodes.unshift(list); // If the list wasn't processed, add it back for further processing
+  }
+}
+
+/**
+ * Adds a description to the section.
+ * @param {import('../types.d.ts').Section} section - The section to add description to.
+ * @param {Array} nodes - The AST nodes.
+ */
+function addDescription(section, nodes) {
+  if (!nodes.length) return;
+
+  if (section.desc) {
+    section.shortDesc = section.desc;
+  }
+
+  const html = getRemarkRehype();
+  const rendered = html.stringify(
+    html.runSync({ type: 'root', children: nodes })
+  );
+  section.desc = rendered || undefined;
+}
+
+/**
+ * Adds additional metadata to the section based on its type.
+ * @param {import('../types.d.ts').Section} section - The section to update.
+ * @param {import('../types.d.ts').Section} parentSection - The parent section.
+ * @param {import('../../types.d.ts').NodeWithData} headingNode - The heading node.
+ */
+function addAdditionalMetadata(section, parentSection, headingNode) {
+  if (!section.type) {
+    section.name = section.textRaw.toLowerCase().trim().replace(/\s+/g, '_');
+    section.displayName = headingNode.data.name;
+    section.type = parentSection.type === 'misc' ? 'misc' : 'module';
+  }
+}
+
+/**
+ * Adds the section to its parent section.
+ * @param {import('../types.d.ts').Section} section - The section to add.
+ * @param {import('../types.d.ts').Section} parentSection - The parent section.
+ */
+function addToParent(section, parentSection) {
+  const pluralType = sectionTypePlurals[section.type];
+  parentSection[pluralType] = parentSection[pluralType] || [];
+  parentSection[pluralType].push(section);
+}
+
+/**
+ * Promotes children to top-level if the section type is 'misc'.
+ * @param {import('../types.d.ts').Section} section - The section to promote.
+ * @param {import('../types.d.ts').Section} parentSection - The parent section.
+ */
+const makeChildrenTopLevelIfMisc = (section, parentSection) => {
+  if (section.type !== 'misc' || parentSection.type === 'misc') {
+    return;
+  }
+
+  Object.keys(section).forEach(key => {
+    if (['textRaw', 'name', 'type', 'desc', 'miscs'].includes(key)) {
+      return;
+    }
+    if (parentSection[key]) {
+      parentSection[key] = Array.isArray(parentSection[key])
+        ? parentSection[key].concat(section[key])
+        : section[key];
+    } else {
+      parentSection[key] = section[key];
+    }
+  });
+};
+
+/**
+ * Handles an entry and updates the parent section.
+ * @param {import('../types.d.ts').HierarchizedEntry} entry - The entry to handle.
+ * @param {import('../types.d.ts').Section} parentSection - The parent section.
+ */
+function handleEntry(entry, parentSection) {
```

Comment:
FYI, I think this util file has way too many methods, I believe you could break them down to specific pieces, like the part of (IMO)